docs/book/src/user/quick-start.md

```diff
@@ -0,0 +1,152 @@
+```
+
+#### Install the Bootstrap Provider Components
+
+{{#tabs name:"tab-installation-bootstrap" tabs:"Kubeadm"}}
+{{#tab Kubeadm}}
+
+```bash
+kubectl create -f https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/releases/latest/download/bootstrap-components.yaml
+```
+
+{{#/tab }}
+{{#/tabs }}
+
+
+#### Install Infrastructure Provider Components
+
+{{#tabs name:"tab-installation-infrastructure" tabs:"AWS"}}
+{{#tab AWS}}
+
+For more information about prerequisites, credentials management, and or permissions for AWS, visit the [getting started guide](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/getting-started.md).
+
+```bash
+kubectl create -f https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/releases/latest/download/infrastructure-components.yaml
+```
+
+{{#/tab }}
+{{#/tabs }}
+
+## Usage
+
+The Cluster API resources are now installed.
+
+![](../images/management-cluster.svg)
+
+Now that we've got Cluster API, Bootstrap and Infrastructure resources installed,
+let's proceed to create a single node cluster.
+
+For the purpose of this tutorial, we'll name our cluster `capi-quickstart`.
+
+{{#tabs name:"tab-usage-cluster-resource" tabs:"AWS"}}
+{{#tab AWS}}
+
+```yaml
+---
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: Cluster
+metadata:
+  name: capi-quickstart
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: AWSCluster
+    name: capi-quickstart
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: AWSCluster
+metadata:
+  name: capi-quickstart
+spec:
+  region:
+  sshKeyName: default
+```
+{{#/tab }}
+{{#/tabs }}
+
+Now that we've created the cluster object, we can create the first control plane Machine.
+
+{{#tabs name:"tab-usage-controlplane-resource" tabs:"AWS"}}
+{{#tab AWS}}
+
+```yaml
+---
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: Machine
+metadata:
+  name: capi-quickstart-controlplane-0
+  labels:
+    cluster.x-k8s.io/control-plane: "true"
+    cluster.x-k8s.io/cluster-name: "capi-quickstart"
+spec:
+  version: v1.15.3
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: capi-quickstart-controlplane-0
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: AWSMachine
+    name: capi-quickstart-controlplane-0
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: AWSMachine
+metadata:
+  name: capi-quickstart-controlplane-0
+spec:
+  instanceType: t2.
+  iamInstanceProfile: "controllers.cluster-api-provider-aws.sigs.k8s.io"
+  sshKeyName: default
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: capi-quickstart-controlplane-0
+spec:
+  initConfiguration:
+    nodeRegistration:
+      name: '{{ ds.meta_data.hostname }}'
```

Comment:
Maybe a comment as to why this is necessary.

Reply:
These details and all the other down should really be in an advanced section of the kubeadm provider, or in another section of this tutorial. Not sure if they're really helpful for this quickstart.

Reply:
If we don't provide some context then users will be confused as well.

Reply:
Especially with this one since it's an embedded jinja template in an otherwise normal yaml file.

Reply:
I tried to write up something about it, but I didn't come up with anything useful, do you have any suggestion? I'm really torn on this because I don't want to make users worry about something that they don't have to mind at this point of their journey in CAPI.

@timothysc any thoughts?

Reply:
Added a note on all the kubeadm specific configuration to look at the kubeadm docs

Reply:
This isn't really kubeadm specific, though...  This is very AWS specific in that we need the Node Name to match with a specific value that is used for looking up the instance within the cloud-provider apis when the cloud-provider integration is enabled.